core/declaration.ts

```diff
@@ -0,0 +1,43 @@
+import { IColumnsDescriptor, mapToColumnProtoArray, Session } from "@dataform/core/session";
+import { dataform } from "@dataform/protos";
+
+export interface DConfig {
+  description?: string;
+  columns?: IColumnsDescriptor;
```

Comment:
Do we want to include `schema` here?  - I see this one related to https://github.com/dataform-co/dataform/issues/357.

My idea in #337 was to add it in _operations_, _tables_, _assertions_. 

Also on L21, add:

```
    if (config.schema) {
      this.schema(config.schema);
    }
```
and then add this:

```
public schema(schema: string) {
   this.proto.target.schema = schema;
}
```

Reply:
I have added `schema` as a field in `DConfig`, because that's absolutely needed here.

I haven't done the other suggestions because:
- I don't think we should expose a `schema()` method, users should be required to fully-qualify schema & name when they call `declare()` in the JS API
- the `schema` field is actually already set at a higher-level than this, inside `Session.declare()` itself